content/en/blog/_posts/2024-02-22-diy-create-your-own-cloud-with-kubernetes-part-1/index.md

```diff
@@ -0,0 +1,158 @@
+
+Have you ever thought about building your own cloud? I bet you have. But is it possible to do this using only modern technologies and approaches, without leaving the cozy Kubernetes ecosystem? Our experience in developing Cozystack required us to delve deeply into it.
+
+You might argue that Kubernetes is not intended for this purpose and why not simply use OpenStack for bare metal servers and run Kubernetes inside it as intended. But by doing so, you would simply shift the responsibility from your hands to the hands of OpenStack administrators. This would add at least one more huge and complex system to your ecosystem.
+
+Why complicate things? - after all, Kubernetes already has everything needed to run tenant Kubernetes clusters at this point.
+
+I want to share with you our experience in developing a cloud platform based on Kubernetes, highlighting the open-source projects that we use ourselves and believe deserve your attention.
+
+In this series of articles, I will tell you our story about how we prepare managed Kubernetes from bare metal using only open-source technologies. Starting from the basic level of data center preparation, running virtual machines, isolating networks, setting up fault-tolerant storage to provisioning full-featured Kubernetes clusters with dynamic volume provisioning, load balancers, and autoscaling.
+
+With this article, I start a series consisting of several parts:
+
+- **Part 1**: Preparing the groundwork for your cloud. Challenges faced during the preparation and operation of Kubernetes on bare metal and a ready-made recipe for provisioning infrastructure.
+- **Part 2**: Networking, storage, and virtualization. How to turn Kubernetes into a tool for launching virtual machines and what is needed for this.
+- **Part 3**: Cluster API and how to start provisioning Kubernetes clusters at the push of a button. How autoscaling works, dynamic provisioning of volumes, and load balancers.
+
+I will try to describe various technologies as independently as possible, but at the same time, I will share our experience and why we came to one solution or another.
+
+To begin with, let's understand the main advantage of Kubernetes and how it has changed the approach to using cloud resources.
+
+It is important to understand that the use of Kubernetes in the cloud and on bare metal differs.
+
+### Kubernetes in the Cloud
+
+When you operate Kubernetes in the cloud, you don't worry about persistent volumes, cloud load balancers, or the process of provisioning nodes. All of this is handled by your cloud provider, who accepts your requests in the form of Kubernetes objects. In other words, the server side is completely hidden from you, and you don't really want to know how exactly the cloud provider implements as it's not in your area of responsibility.
+
+![kubernetes in the cloud illustration](cloud.svg)
```

Comment:
```suggestion
![A diagram showing cloud Kubernetes, with load balancing done outside the cluster](cloud.svg)
```
Better: use a [figure](https://gohugo.io/content-management/shortcodes/#figure) shortcode and add a caption. Also keep the alt attribute for accessibility.